apisix/core/ctx.lua

```diff
@@ -274,6 +276,22 @@ do
                     end
                 end
 
+            elseif core_str.has_prefix(key, "json_body_arg_") then
+                -- only match json type request body
+                local content_type = request.header(nil, "Content-Type")
+                if content_type ~= nil and core_str.has_prefix(content_type,
+                        "application/json") then
+                    local arg_key = sub_str(key, 15)
+                    local raw_request_body = request.get_body()
+                    local body, err = json.decode(raw_request_body)
```

Comment:
Consider using a cache, as this code may be executed multiple times on hot paths.